std/format/read.d

```diff
@@ -692,6 +693,99 @@ if (isSomeString!(typeof(fmt)))
     assert(aa2 == ["hello":1, "world":2]);
 }
 
+/**
+Reads an input range according to a format string and returns a tuple with the
+read values.
+
+Format specifiers with format character $(B 'd'), $(B 'u') and $(B
+'c') can take a $(B '*') parameter for skipping values.
+
+The second version of `formattedRead` takes the format string as
+template argument. In this case, it is checked for consistency at
+compile-time.
+
+Params:
+    Args = a variadic list of types of the arguments
+ */
+template formattedRead(Args...)
+if (Args.length && allSatisfy!(isType, Args))
+{
+    import std.typecons : Flag, Tuple, Yes;
+
+    /**
+    Params:
+        r = an $(REF_ALTTEXT input range, isInputRange, std, range, primitives),
+            where the formatted input is read from
+        fmt = a $(MREF_ALTTEXT format string, std,format)
+        Range = the type of the input range `r`
+        Char = the character type used for `fmt`
+
+    Returns:
+        A Tuple!Args with the elements filled. If the input range `r` ends early,
+        the missing arguments will be default initialized.
+
+    Throws:
+        A $(REF_ALTTEXT FormatException, FormatException, std, format)
+        if reading did not succeed.
+    */
+    Tuple!Args formattedRead(Range, Char)(auto ref Range r, const(Char)[] fmt)
+    {
+        import core.lifetime : forward;
+        import std.exception : enforce;
+        import std.format : FormatException;
+
+        Tuple!Args args;
+        const numArgsFilled = .formattedRead(forward!r, fmt, args.expand);
+        enforce!FormatException(numArgsFilled == Args.length);
```

Comment:
1. `enforce!FormatException` should be replaced with the alias `enforceFmt` for consistency with other code in this package.
2. Please add a descriptive error message to the exception.

Reply:
done

Reply:
Oops, done